source/fe/mapping_manifold.cc

```diff
@@ -213,14 +213,18 @@ MappingManifold<dim,spacedim>::
 transform_unit_to_real_cell (const typename Triangulation<dim,spacedim>::cell_iterator &cell,
                              const Point<dim> &p) const
 {
-  std::vector<Point<spacedim> > vertices;
-  std::vector<double> weights;
+  std::array<Point<spacedim>, GeometryInfo<dim>::vertices_per_cell> vertices;
+  std::array<double, GeometryInfo<dim>::vertices_per_cell> weights;
+
   for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
     {
-      vertices.push_back(cell->vertex(v));
-      weights.push_back(GeometryInfo<dim>::d_linear_shape_function(p,v));
+      vertices[v] = cell->vertex(v);
+      weights[v] = GeometryInfo<dim>::d_linear_shape_function(p, v);
     }
-  return cell->get_manifold().get_new_point(vertices, weights);
+  return cell->get_manifold().get_new_point(make_array_view(vertices.begin(),
+                                                            vertices.end()),
+                                            make_array_view(weights.begin(),
+                                                            weights.end()));
```

Comment:
Should we have another version of `make_array_view` that just takes a `std::array` as argument? (Just like we already have one for `std::vector`.)

Reply:
I could go either way on this one, but I have a slight preference towards using the same iterator-based template that works with all array classes instead of writing half a dozen wrappers.